ui/src/utils/date.ts

```diff
@@ -12,4 +12,25 @@ export function shortDate(date: Date): string {
     dateStyle: 'short',
     timeStyle: 'medium',
   }).format(date);
-}
\ No newline at end of file
+}
+
+export function formatMilliseconds(durationInMs): string {
+  if (durationInMs >= 3600000) { // 1 hour = 3600000 ms
+    const hours = Math.floor(durationInMs / 3600000);
+    durationInMs %= 3600000;
+    
+    const minutes = Math.floor(durationInMs / 60000);
+    return `${hours}h ${minutes}m`;
+  } else if (durationInMs >= 60000) { // 1 minute = 60000 ms
+    const minutes = Math.floor(durationInMs / 60000);
+    durationInMs %= 60000;
+
+    const seconds = Math.floor(durationInMs / 1000);
+    return `${minutes}m ${seconds}s`;
+  } else if (durationInMs >= 1000) { // 1 second = 1000 ms
+    const seconds = Math.floor(durationInMs / 1000);
+    return `${seconds}s`;
+  } else {
+    return `${durationInMs}ms`;
+  }
+}
```

Comment:
This would be more readable using constants

```suggestion
const second = 1000;
const minute = 60 * second;
const hour = 60 * minute;

export function formatMilliseconds(durationInMs): string {
  if (durationInMs >= hour) {
    const hours = Math.floor(durationInMs / hour);
    durationInMs %= hour;
    
    const minutes = Math.floor(durationInMs / minute);
    return `${hours}h ${minutes}m`;
  } else if (durationInMs >= minute) {
    const minutes = Math.floor(durationInMs / minute);
    durationInMs %= minute;

    const seconds = Math.floor(durationInMs / second);
    return `${minutes}m ${seconds}s`;
  } else if (durationInMs >= second) {
    const seconds = Math.floor(durationInMs / second);
    return `${seconds}s`;
  } else {
    return `${durationInMs}ms`;
  }
}
```